test/replication-luatest/gh_9720_tmp_spaces_ddl_abort_applier_txs_test.lua

```diff
@@ -0,0 +1,166 @@
+local t = require('luatest')
+local replica_set = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group(nil, t.helpers.matrix({memtx_use_mvcc = {false, true}}))
+
+g.before_each(function(cg)
+    cg.replica_set = replica_set:new{}
+    local box_cfg = {
+        memtx_use_mvcc_engine = cg.params.memtx_use_mvcc,
+        replication = {
+            server.build_listen_uri('server1', cg.replica_set.id),
+            server.build_listen_uri('server2', cg.replica_set.id),
+        },
+        replication_timeout = 0.1,
+    }
+    cg.server1 = cg.replica_set:build_and_add_server{
+        alias = 'server1',
+        box_cfg = box_cfg,
+    }
+    box_cfg.read_only = true
+    cg.server2 = cg.replica_set:build_and_add_server{
+        alias = 'server2',
+        box_cfg = box_cfg,
+    }
+    cg.replica_set:start()
+    cg.server1:exec(function()
+        local s = box.schema.create_space('s')
+        s:create_index('p')
+        box.begin()
+        for i = 1, 1000 do
+            s:replace{i, i}
+        end
+        box.commit()
+    end)
+    cg.server2:exec(function()
+        box.cfg{read_only = false}
+    end)
+    cg.replica_set:wait_for_fullmesh()
+    cg.server1:wait_for_downstream_to(cg.server2)
+end)
+
+g.after_each(function(cg)
+    cg.replica_set:drop()
+end)
+
+-- Test that a DDL transaction on fully temporary spaces does not abort a remote
+-- transaction.
+g.test_fully_temporary_ddl_does_not_abort_remote_tx = function(cg)
+    t.tarantool.skip_if_not_debug()
+
+    cg.server2:exec(function()
+        box.error.injection.set('ERRINJ_CHECK_FORMAT_DELAY', true)
+    end)
+    cg.server1:exec(function()
+        box.space.s:format{'i', 'unsigned'}
+    end)
+    cg.server2:exec(function()
+        box.schema.space.create('tmp', {type = 'temporary'})
+        box.error.injection.set('ERRINJ_CHECK_FORMAT_DELAY', false)
+    end)
```

Comment:
Does it fail without the patch? It looks like the last exec is likely to be executed before the second one is replicated. So the format coming to `server2` won't even see `ERRINJ_CHECK_FORMAT_DELAY = true` and won't try to abort anything.

Reply:
Yes, it fails. I see your point, but how do you suggest setting up a synchronization point here?

I tried to make the `_space` synchronous and wait for `box.info.synchro.queue.busy` on `server2`, but the replication connection starts to timeout then:
```console
2024-08-03 12:15:11.398 [42760] relay/unix/:(socket)/101/main coio.c:300 E> TimedOut: timed out
2024-08-03 12:15:11.398 [42760] relay/unix/:(socket)/101/main I> exiting the relay loop
```

Reply:
Easy but not reliable way - just add here a `fiber.sleep(0.01)` to make it more likely to fail without the patch. Hard, but more correct way - introduce a errinj like `ERRINJ_WAL_DELAY_COUNTDOWN` or other `*_COUNTDOWN`s. An injection which gets set on Nth check. Such as `ERRINJ_CHECK_FORMAT_DELAY_COUNTDOWN`. And make it set `ERRINJ_CHECK_FORMAT_DELAY`. Then you can set countdown = 0, and wait until delay = true.

Reply:
Thanks, added a `ERRINJ_CHECK_FORMAT_DELAY_COUNTDOWN` injection.